lib/server-builder.js

```diff
@@ -0,0 +1,160 @@
+    args.push('assembleAndroidTest');
+
+    return {cmd, args};
+  }
+
+  async setGradleWrapperVersion (version) {
+    const propertiesPath = path.resolve(this.serverPath, 'gradle', 'wrapper', 'gradle-wrapper.properties');
+    const originalProperties = await fs.readFile(propertiesPath, 'utf8');
+    const newProperties = this.updateGradleDistUrl(originalProperties, version);
+    await fs.writeFile(propertiesPath, newProperties, 'utf8');
+  }
+
+  updateGradleDistUrl (propertiesContent, version) {
+    return propertiesContent.replace(
+      new RegExp(`^(${_.escapeRegExp(GRADLE_URL_PREFIX)}).+$`, 'gm'),
+      `$1${GRADLE_URL_TEMPLATE.replace('VERSION', version)}`
+    );
+  }
+
+  async insertAdditionalAppDependencies (additionalAppDependencies) {
+    const buildPath = path.resolve(this.serverPath, 'app', 'build.gradle');
+    const originalConfiguration = await fs.readFile(buildPath, 'utf8');
+    const newConfiguration = this.updateDependencyLines(originalConfiguration, additionalAppDependencies);
+    await fs.writeFile(buildPath, newConfiguration, 'utf8');
+  }
+
+  updateDependencyLines (configurationContent, additionalAppDependencies) {
+    const dependencyLines = additionalAppDependencies
+      .map(function (dependency) {
+        // Disallow whitespace and quote characters that can break string literals in a patched build.gradle
+        // and dollar characters that otherwise would be interpreted by String.replace below
+        if (/[\s'\\$]/.test(dependency)) {
+          throw new Error('Single quotes, dollar characters and whitespace characters' +
+            ` are disallowed in additional dependencies: ${dependency}`);
+        }
+        return `implementation '${dependency}'`;
+      })
+      .join('$1'); // interpreted as Group 1 from the pattern below
+
+    return configurationContent.replace(
+      // Group 1 captures new line characters and indentation (eg. '\n\t') used in build.gradle file
+      // This ensures that a patched build.gradle will have correct new lines and indentation
+      new RegExp(`(\\s*^\\s*)${_.escapeRegExp(GRADLE_DEPENDENCIES_PLACEHOLDER)}\\s*$`, 'gm'),
+      `$1${dependencyLines}`
+    );
+  }
+
+  async runBuildProcess () {
+    const {cmd, args} = this.getCommand();
+    log.debug(`Beginning build with command '${cmd} ${args.join(' ')}' ` +
+      `in directory '${this.serverPath}'`);
+    const gradlebuild = new SubProcess(cmd, args, {
+      cwd: this.serverPath,
+      stdio: ['ignore', 'pipe', 'pipe'],
+    });
+    let buildLastLines = [];
+
+    const logMsg = `Output from Gradle ${this.showGradleLog ? 'will' : 'will not'} be logged`;
+    log.debug(`${logMsg}. To change this, use 'showGradleLog' desired capability`);
+    gradlebuild.on('stream-line', line => {
+      // if we have an error we want to output the logs
+      // otherwise the failure is inscrutible
+      // but do not log expected errors
+      const ignoreError = IGNORED_ERRORS.some((x) => line.includes(x));
```

Comment:
Why do we keep IGNORED_ERRORS array empty?

Reply:
For possible future filtering. Removed it now.

Reply:
YAGNI ;)